department-of-veterans-affairs/caseflow
# TYLERB/APPEALS-8779: Implement tab routing for CSP task actions

Resolves [APPEALS-8779](https://vajira.max.gov/browse/APPEALS-8779)

### Description
Change routing for a few task actions for Vha Caregiver Support. Verify that the user is redirected to the appropriate tab according to the designs

### Acceptance Criteria
- [ ] The Mark task as in progress task routes to the In Progress tab after modal submission
- [ ] The Documents ready for Board Intake review task routes to the Completed tab  after modal submission
- [ ] The Return to Board Intake task routes to the Completed tab after modal submission

### Testing Plan

1. Create some tasks to use as test data

```ruby
#create some tasks
3.times do
  appeal = FactoryBot.create(:appeal, :with_pre_docket_task)
  parent_task = appeal.tasks.last
  caregiver_task = VhaDocumentSearchTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, 
  assigned_to: VhaCaregiverSupport.singleton )
end
```

2. Go to the unassigned tab, select one of the new tasks, and then perform the mark task in progress task action.

3. Verify that the user is routed to the in progress tab after task modal submission

4. Select the user that was just moved to in progress tab. Then perform the ready for board intake review task action.

5. Verify that the user is routed to the completed tab after task modal submission 

6. Return back to the unassigned tab, select one of the new tasks, and then perform the return to board intake task action.

7. Verify that the user is routed to the completed tab after task modal submission.
